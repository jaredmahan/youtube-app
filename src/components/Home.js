import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { Search } from "../components/Search";
import { Result } from "../components/Result";
import { Loading } from "../components/Loading";

const List = styled.div`
  & > div:nth-child(even) {
    background-color: #f7f7f7;
  }
`;

export function Home(props) {
  return (
    <div>
      <Search
        order={props.order}
        queryChange={props.queryChange}
        orderChange={props.orderChange}
      />
      {props.errorMessage && <em>{props.errorMessage}</em>}
      <Loading loading={props.loading} loadingText="Loading your results...">
        {!!props.results && (
          <List>
            {props.results.length > 0 &&
              props.results.map((result, index) => (
                <Result key={index} result={result} />
              ))}
          </List>
        )}
        {!!props.results &&
          (props.results.length < 0 && (
            <em>No results found based on your query.</em>
          ))}
      </Loading>
    </div>
  );
}

Home.propTypes = {
  query: PropTypes.string,
  order: PropTypes.string,
  queryChange: PropTypes.func,
  orderChange: PropTypes.func
};

export default Home;
