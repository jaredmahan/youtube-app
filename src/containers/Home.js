import React, { Component } from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import * as actions from "../actions/home";
import { connect } from "react-redux";
import { Home } from "../components/Home";

export class HomeContainer extends Component {
  render() {
    return (
      <Home
        {...this.props}
        orderChange={this.handleOrderChange}
        queryChange={this.handleQueryChange}
      />
    );
  }

  handleQueryChange = e => {
    const value = !!e && !!e.target ? e.target.value.trim() : null;
    this.props.actions.getYouTubeResults(value, this.props.order);
  };
  handleOrderChange = e => {
    const value = !!e && !!e.target ? e.target.value.trim() : null;
    this.props.actions.getYouTubeResults(this.props.query, value);
  };
}

Home.propTypes = {
  actions: PropTypes.object.isRequired,
  errorMessage: PropTypes.string,
  order: PropTypes.string,
  query: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired
};

function mapStateToProps(state) {
  return {
    loading: state.home.loading,
    order: state.home.order,
    errorMessage: state.home.errorMessage,
    results: !!state.home.result ? state.home.result.items : null,
    query: state.home.query
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeContainer);
