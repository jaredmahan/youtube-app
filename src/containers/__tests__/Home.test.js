import React from "react";
import { HomeContainer } from "../Home";
import ShallowRenderer from "react-test-renderer/shallow";

describe("<HomeContainer>", () => {
  it("should render correctly", () => {
    const renderer = new ShallowRenderer();
    const props = {
      query: "query",
      order: "order",
      loading: false,
      results: [{ id: 1 }, { id: 2 }, { id: 3 }],
      actions: {}
    };
    renderer.render(<HomeContainer {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it("handleOrderChange should call the getYouTubeResults action", () => {
    const props = {
      actions: {
        getYouTubeResults: jest.fn()
      }
    };
    const component = new HomeContainer(props);
    component.handleOrderChange();
    expect(props.actions.getYouTubeResults).toBeCalledTimes(1);
  });
  it("handleQueryChange should call the getYouTubeResults action", () => {
    const props = {
      actions: {
        getYouTubeResults: jest.fn()
      }
    };
    const component = new HomeContainer(props);
    component.handleQueryChange();
    expect(props.actions.getYouTubeResults).toBeCalledTimes(1);
  });
});
