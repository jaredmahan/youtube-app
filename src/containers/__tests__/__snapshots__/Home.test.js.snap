// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeContainer> should render correctly 1`] = `
<Home
  actions={Object {}}
  loading={false}
  order="order"
  orderChange={[Function]}
  query="query"
  queryChange={[Function]}
  results={
    Array [
      Object {
        "id": 1,
      },
      Object {
        "id": 2,
      },
      Object {
        "id": 3,
      },
    ]
  }
/>
`;
